src/acquisition/covidcast/database.py

```diff
@@ -278,12 +281,54 @@ def get_covidcast_meta(self):
       ORDER BY
         t.`time_type` ASC,
         t.`geo_type` ASC
-      '''
-      self._cursor.execute(sql, (source, signal))
-      meta.extend(list(dict(zip(self._cursor.column_names,x)) for x in self._cursor))
+    '''
+
+    meta = []
+    meta_lock = threading.Lock()
+
+    def worker():
+      try:
+        print("starting thread: " + threading.current_thread().name)
+        #  set up new db connection for thread
+        worker_dbc = Database()
+        worker_dbc.connect(connector_impl=self._connector_impl)
+        w_cursor = worker_dbc._cursor
+        while True:
+          (source, signal) = srcsigs.get_nowait() # this will throw the Empty caught below
+          w_cursor.execute(inner_sql, (source, signal))
+          with meta_lock:
+            meta.extend(list(
+              dict(zip(w_cursor.column_names, x)) for x in w_cursor
+            ))
+          srcsigs.task_done()
+      except Empty:
+        print("no jobs left, thread terminating: " + threading.current_thread().name)
+      finally:
+        worker_dbc.disconnect(False) # cleanup
+
+    threads = []
+    for n in range(n_threads):
+      t = threading.Thread(target=worker, name='MetacacheThread-'+str(n))
+      t.start()
+      threads.append(t)
+
+    srcsigs.join()
+    print("jobs complete")
```

Comment:
fwiw if you want to persist this to logs (and be able to view in kibana) you can follow the approach in covidcast_meta_cache_updater.py

Reply:
cool, ill check it out